include/boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp

```diff
@@ -1197,7 +1197,7 @@ struct buffered_piece_collection
         robust_point_type any_point;
         geometry::recalculate(any_point, point, m_robust_policy);
 
-        int count_in_original = 0;
+        signed_index_type count_in_original = 0;
```

Comment:
Why it should stay `int`? It's used to count the elements of a vector meeting some criteria. This means that it should be at least of size of `std::vector<robust_original>::size_type`. But since it can also be decremented, to be perfectly safe it could be replaced by 2 variables, one counting the elements meeting the `m_is_interior` condition and the other one counting the elements meeting the `m_has_interior` condition. And at the end both values should be compared which one is greater. Do you agree?